### PR TITLE
App doesn't take you to a Space after choosing to Join it

### LINKED ISSUE #3933: App doesn't take you to a Space after choosing to Join it
### Steps to reproduce

1. Sent a URL invite to Space in DM (currently in Home space, looking at DM)
2. Click on URL > Bottom sheet appears
3. Click Join...

### What did you expect?

I expected to be taken to the Space I had just joined.

### What happened?

Space joined but no UI change

### Your phone model

_No response_

### Operating system version

_No response_

### Application version and app store

https://buildkite.com/organizations/matrix-dot-org/pipelines/element-android/builds/3224/jobs/a1fe572d-8a7b-4406-8a0f-37f4c9ff4787/artifacts/30990f61-8f7d-4dc6-af1e-340765110cc6

### Homeserver

_No response_

### Have you submitted a rageshake?

No

### PR DESCRIPTION
**Pull request by BillCarsonFr**
Fixes #3933 

The bottomsheet listener was using a weak reference. Based on the assumption that those listener would probably be activities. But it's not the case as the default navigator is instanciating one by default.

Also the listener way is not resistent to configuration change, moved it to fragment callback insted

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailActivity.kt [hunk 1]
```diff
@@ -32,25 +34,44 @@ import im.vector.app.core.platform.ToolbarConfigurable
...
         RequireActiveMembershipViewModel.Factory,
-        RoomWidgetPermissionViewModel.Factory {
+        RoomWidgetPermissionViewModel.Factory,
+        MatrixToBottomSheet.InteractionListener {
 
     override fun getBinding(): ActivityRoomDetailBinding {
         return ActivityRoomDetailBinding.inflate(layoutInflater)
     }
 
+    private val fragmentLifecycleCallbacks = object : FragmentManager.FragmentLifecycleCallbacks() {
```
Comment (ouchadam) [inline vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailActivity.kt:58]: Suggestion: could this be extracted and reused? 

maybe something like this~

```kotlin
class MatrixBottomSheetCallbacks(
        private val provider: () -> MatrixToBottomSheet.InteractionListener
) : FragmentManager.FragmentLifecycleCallbacks() {
    override fun onFragmentResumed(fm: FragmentManager, f: Fragment) {
        if (f is MatrixToBottomSheet) {
            f.interactionListener = provider()
        }
    }

    override fun onFragmentPaused(fm: FragmentManager, f: Fragment) {
        if (f is MatrixToBottomSheet) {
            f.interactionListener = null
        }
    }
}

// activity
private val bottomSheetCallbacks = MatrixBottomSheetCallbacks { this }
```

Comment (bmarty) [reply vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailActivity.kt:58]: We also use SharedViewModel to let Fragments communicate between each others or with their host Activity

**Review by bmarty — APPROVED** [review]
LGTM

File: vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailActivity.kt (hunk 1 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailActivity.kt:58]: @ouchadam feel free to open a dedicated PR to implement your suggestion :)

**Pull request merged**